code/mob/living.dm

```diff
@@ -2013,3 +2014,16 @@ var/global/icon/human_static_base_idiocy_bullshit_crap = icon('icons/mob/human.d
 			if(thr?.user)
 				src.was_harmed(thr.user, AM)
 	..()
+
+/mob/living/proc/check_singing_prefix(var/message)
+	if (isalive(src))
+		// check for "%"
+		if (dd_hasprefix(message, singing_prefix))
+			src.singing = NORMAL_SINGING
+			return copytext(message, 2)
+		// check for " %"
```

Comment:
couldn't you just trim the message?

Reply:
Could you give me an example of what you mean?

Reply:
Do you mean using trim()? I think that would remove the "%" but I need to read that.

Reply:
I think trim() just removes whitespace, unless I'm wrong...

Reply:
Looks like it removes all characters with ascii code > 32
![image](https://user-images.githubusercontent.com/7035767/97514670-710ed480-19e3-11eb-87ce-21a8ede10cdd.png)

Reply:
Oh I think you're right

Reply:
I've changed it so that say() is called before checks for '%'